Fb input interp

RobertoPadilla-NOAA opened the pull request:
This is the branch to interpolate fields from regular grids to a curvilinear grids

ajhenrique committed:
Hotfix/hf master (#45) (#54)

Pull request for HF to master featuring:
* bugfix for netcdf file id allocation
* remove extra line due to copy/paste error
* add WCOR to switch list
* add check on array index
* undo modification on ndates
* add feature for fixed filename with ndates=0 for ww3_ounf

ajhenrique committed:
HF_slurm_regtests

Timing option -T added to run_test, option to run batch queue [slurm] -b added to run_test, created new matrix_ncep with slurm directives for R&D testing, corrected bug in matrix.base ww3_ts4 [mpi run were missing -p -n flags. (#67)(cherry picked from commit 972106c8af8b72cdd54aada91908856934912f06)

ajhenrique committed:
Branch FB_InputInterp: preparing general arrays and loop blocks for generalized interpolation to curvilinear grids.

Commit:
Adding interpolation code and functions in wmupdtmd.ftn

Commit:
adding interpolation function and subroutine in wmupdtmd.ftn

Commit:
Adding Func & Subr to interp recti to curvi grids

nacbrother committed:
Corrected th2m and sth2m calculations. (#103)

On lines 1439 and 1440, the calculations for ABX2 and ABY2 were like that of ABX and ABY.  This may have been an oversight when inserted.

ajhenrique committed:
Revert "Corrected th2m and sth2m calculations. (#103)" (#111)

This reverts commit 35040048983087cc76ab3858a747a356625e917f.

Commit:
Solving bugs in wmupdtmd.ftn for curvi interp

Commit:
Cleaning the code  wmupdtmd.ftn for curvi interp

Commit:
wmupdtmd.ftn eddited, ready for push

Commit:
Merge remote-tracking branch 'origin/develop' into FB_InputInterp

updating FB_InputInterp from upstream and develop branch

ajhenrique commented:
> In SUBROUTINE W3EXNC the parameter NCIDS should then be declared as:
> INTEGER, INTENT(INOUT) :: NCIDS(NOGRP,NGRPP,NOSWLL + 1)
> The present code assumes NOSWLL == 5:
> 
> https://github.com/NOAA-EMC/WW3/blob/39d1fb73d8ffde828e3070eafc49f6fd5951dfdb/model/ftn/ww3_ounf.ftn#L760

@CarstenHansen does this comment belong here?

ajhenrique committed:
Merge branch 'master' into FB_InputInterp

ajhenrique committed:
Update ww3_ounf.ftn

Making INTEGER, INTENT(INOUT) :: NCIDS(NOGRP,NGRPP,NOSWLL + 1) allows for NOSWLL different to the default =5, which can be the case when using the NCO switch (NOSWLL=3). Bugfix suggested by @CarstenHansen

ajhenrique committed:
Update ww3_ounf.nml

ajhenrique reviewed (approved):
Changes passed NCEP regtest matrix and comparison of runtime and outputs against file-based input to curvilinear grids. Will require adjustments to account for differences in ice mask reconciliation that will be introduced in a later commit.
